streams/src/main/java/org/apache/kafka/streams/kstream/internals/suppress/SuppressedImpl.java

```diff
@@ -0,0 +1,283 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream.internals.suppress;
+
+import org.apache.kafka.streams.kstream.Suppressed;
+import org.apache.kafka.streams.kstream.Windowed;
+import org.apache.kafka.streams.processor.ProcessorContext;
+
+import java.time.Duration;
+import java.util.Objects;
+
+import static org.apache.kafka.streams.kstream.internals.suppress.SuppressedImpl.BufferFullStrategy.SHUT_DOWN;
+
+public final class SuppressedImpl<K> implements Suppressed<K> {
+    private static final TimeDefinition<?> DEFAULT_TIME_DEFINITION = (context, anyKey) -> context.timestamp();
```

Comment:
We can make it a `<K>`, but only if we drop the `static`. The static may be nice for performance, since we only need one function class & instance, but I'm not sure if it matters that much.

Reply:
This is not runtime critical -- and having a few more objects (we don't have about hundreds) is not overhead concern IMHO. It would be cleaner with specify type and get rid of the "cast helper method" IMHO.